rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/db/RMDBManager.java

```diff
@@ -58,31 +59,33 @@
 import org.hibernate.Query;
 import org.hibernate.Session;
 import org.hibernate.SessionFactory;
-import org.hibernate.Transaction;
 import org.hibernate.cfg.Configuration;
 
 
 public class RMDBManager {
 
-    private static final String JAVA_PROPERTYNAME_NODB = "rm.database.nodb";
+    private static final String RM_DATABASE_IN_MEMORY = "rm.database.nodb";
+
     private static final Logger logger = ProActiveLogger.getLogger(RMDBManager.class);
 
     private final SessionFactory sessionFactory;
 
-    private static RMDBManager instance = null;
+    private final TransactionHelper transactionHelper;
 
-    private Timer timer = null;
+    private static class LazyHolder {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a private constructor to hide the implicit public one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1118)